app/controllers/admin/payment_preferences_controller.rb

```diff
@@ -0,0 +1,254 @@
+      update_payment_preferences
+    elsif params[:stripe_api_keys_form].present? && @stripe_available
+      update_stripe_keys
+    end
+
+    redirect_to action: :index
+  end
+
+  private
+
+  MIN_COMMISSION_PERCENTAGE = 0
+  MAX_COMMISSION_PERCENTAGE = 100
+
+  def ensure_payments_enabled
+    @paypal_enabled = PaypalHelper.paypal_active?(@current_community.id)
+    @stripe_available = StripeHelper.stripe_feature_enabled?(@current_community.id)
+    @stripe_enabled = StripeHelper.stripe_provisioned?(@current_community.id)
+    unless @paypal_enabled || @stripe_enabled
+      flash[:error] = t("admin.communities.settings.payments_not_enabled")
+      redirect_to admin_details_edit_path
+    end
+  end
+
+  def paypal_index
+    paypal_account = paypal_accounts_api.get(community_id: @current_community.id).data
+
+    {
+      order_permission_action: admin_paypal_preferences_account_create_path(),
+      paypal_account: paypal_account
+    }
+  end
+
+  def stripe_index
+    stripe_account = stripe_tx_settings
+    {
+      stripe_account: stripe_account,
+      stripe_api_form: StripeApiKeysForm.new
+    }
+  end
+
+  def paypal_tx_settings
+    Maybe(tx_settings_api.get(community_id: @current_community.id, payment_gateway: :paypal, payment_process: :preauthorize))
+    .select { |result| result[:success] }
+    .map { |result| result[:data] }
+    .or_else({})
+  end
+
+  def stripe_tx_settings
+    Maybe(tx_settings_api.get(community_id: @current_community.id, payment_gateway: :stripe, payment_process: :preauthorize))
+    .select { |result| result[:success] }
+    .map { |result| result[:data] }
+    .or_else({})
+  end
+
+  def build_prefs_form(params = nil)
+    currency = @current_community.currency
+
+    if @paypal_enabled
+      minimum_commission = paypal_minimum_commissions_api.get(currency)
+      tx_settings = paypal_tx_settings
+    else
+      minimum_commission = 0
+      tx_settings = stripe_tx_settings
+    end
+
+    form = PaymentPreferencesForm.new(
+      minimum_commission: minimum_commission,
+      commission_from_seller: tx_settings[:commission_from_seller],
+      minimum_listing_price: Money.new(tx_settings[:minimum_price_cents], currency),
+      minimum_transaction_fee: Money.new(tx_settings[:minimum_transaction_fee_cents], currency),
+      marketplace_currency: currency
+    )
+    {payment_prefs_form: form, payment_prefs_valid: form.valid? }
+  end
+
+  def build_view_locals
+    @selected_left_navi_link = "payment_preferences"
+
+    onboarding_popup_locals = OnboardingViewUtils.popup_locals(
+      flash[:show_onboarding_popup],
+      admin_getting_started_guide_path,
+      Admin::OnboardingWizard.new(@current_community.id).setup_status)
+
+    view_locals = {
+      min_commission_percentage: MIN_COMMISSION_PERCENTAGE,
+      max_commission_percentage: MAX_COMMISSION_PERCENTAGE,
+      available_currencies: MarketplaceService::AvailableCurrencies::CURRENCIES,
```

Comment:
solved in https://github.com/sharetribe/sharetribe/pull/3018/commits/afd4f143d612ada4b6608232eeb121fa09734c9f